frontend/app/components/input-file.tsx

```diff
@@ -0,0 +1,98 @@
+import { InputError } from '~/components/input-error';
+import { InputHelp } from '~/components/input-help';
+import { InputLabel } from '~/components/input-label';
+import { cn } from '~/utils/tailwind-utils';
+
+const inputBaseClassName =
+  'w-full sm:w-auto block rounded-sm border-gray-300 bg-gray-200 text-slate-700 hover:bg-neutral-300 focus:bg-neutral-300';
+const inputDisabledClassName =
+  'disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70';
+const inputErrorClassName = 'border-red-500 focus:border-red-500 focus:ring-red-500';
+
+export interface InputFileProps {
+  ariaDescribedby?: string;
+  accept: string;
+  className?: string;
+  defaultValue?: string;
+  errorMessage?: string;
+  helpMessagePrimary?: React.ReactNode;
+  helpMessagePrimaryClassName?: string;
+  helpMessageSecondary?: React.ReactNode;
+  helpMessageSecondaryClassName?: string;
+  id: string;
+  label: string;
+  name: string;
+  required?: boolean;
+}
+
+export function InputFile({
+  ariaDescribedby,
+  errorMessage,
+  className,
+  helpMessagePrimary,
+  helpMessagePrimaryClassName,
+  helpMessageSecondary,
+  helpMessageSecondaryClassName,
+  id,
+  label,
+  required,
+  ...restInputProps
+}: InputFileProps) {
+  const inputErrorId = `input-${id}-error`;
+  const inputHelpMessagePrimaryId = `input-${id}-help-primary`;
+  const inputHelpMessageSecondaryId = `input-${id}-help-secondary`;
+  const inputLabelId = `input-${id}-label`;
+  const inputWrapperId = `input-${id}`;
+
+  function getAriaDescribedby() {
+    const describedby = [];
+    if (ariaDescribedby) describedby.push(ariaDescribedby);
+    if (helpMessagePrimary) describedby.push(inputHelpMessagePrimaryId);
+    if (helpMessageSecondary) describedby.push(inputHelpMessageSecondaryId);
+    return describedby.length > 0 ? describedby.join(' ') : undefined;
+  }
```

Comment:
I think this could be made more concise..

``` ts
function getAriaDescribedby(): string[] | undefined {
  return [
    ariaDescribedby,
    inputHelpMessagePrimaryId, 
    inputHelpMessageSecondaryId
  ].filter(Boolean).join(' ') || undefined;
}
```

Reply:
Not sure this is *better*. If you don't feel that it is, just resolve this comment.